wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -15,26 +15,33 @@
  */
 package com.squareup.wire.gradle
 
+import com.android.builder.model.AndroidProject.FD_GENERATED
+import com.squareup.wire.VERSION
+import com.squareup.wire.gradle.kotlin.sourceRoots
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.api.plugins.JavaBasePlugin
-import org.gradle.api.tasks.SourceSetContainer
+import org.gradle.api.artifacts.UnknownConfigurationException
 import org.gradle.api.tasks.TaskProvider
-import org.jetbrains.kotlin.gradle.plugin.KotlinBasePluginWrapper
+import org.gradle.api.tasks.compile.JavaCompile
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
+import org.jetbrains.kotlin.gradle.plugin.sources.DefaultKotlinSourceSet
 import java.io.File
+import java.util.concurrent.atomic.AtomicBoolean
 
 class WirePlugin : Plugin<Project> {
-  private var kotlin = false
-  private var java = false
+  private val android = AtomicBoolean(false)
+  private val java = AtomicBoolean(false)
+  private val kotlin = AtomicBoolean(false)
 
-  private lateinit var sourceSetContainer: SourceSetContainer
+  private lateinit var extension: WireExtension
+  internal lateinit var project: Project
 
-  override fun apply(project: Project) {
-    val logger = project.logger
+  private val sources by lazy { this.sourceRoots(javaOnly = !kotlin.get() && java.get()) }
 
-    val extension = project.extensions.create(
-        "wire", WireExtension::class.java, project
-    )
+  override fun apply(project: Project) {
+    this.extension = project.extensions.create("wire", WireExtension::class.java, project)
+    this.project = project
```

Comment:
explode if this is already initialized?

Reply:
(I guess I should read what the rules are for this)

Reply:
The line above this will already throw if it's called twice.

It's never called twice, though.

Reply:
I'll leave it as is.